### PR TITLE
Move session params show_log and log_level as a global configuration.

### LINKED ISSUE #7: Print logs/hints about what we are doing at the different stages when launching a session
**Is your feature request related to a problem? Please describe.**

The logs is still doesn't every helpful when error occurs. Could we print something like (when `show_logs=True`):

```
Launching coordinator...
Coordinator service is ready.
Launching graphscope engines... (or, `Waiting graphscope engines....`)
Graphscope engines are ready.
```

I mean, we should let user know what we are doing or waiting for, in a detailed manner, when `show_logs` is `True`.

**Additional context**

I just think log in comments https://github.com/alibaba/GraphScope/issues/3#issuecomment-747198036 is quite confusing. The word "ready" combined with no extra logs deliver a sense of "we are ready", but actually things are not ready, and users are confused about what they are waiting for and why it doesn't return or raise.

### LINKED ISSUE #27: Make `show_log` as a global configuration of graphscope.
**Is your feature request related to a problem? Please describe.**
The `show_log` parameters between different sessions will affect each other.

**Describe the solution you'd like**

Deprecated `show_log`, `log_level` params in the session,  instead，It is better to define as global configuration of graphscope.

```python
import graphscope
graphscope.set_option("show_log", True)
graphscope.get_option("show_log")
```

### PR DESCRIPTION
**Pull request by lidongze0629**
## What do these changes do?

 - Print the process information when launching session
 - Dump session status when launch failed even show_log is False
 - Move `show_log` and `log_level` as a global configuration.
 - Add graphscope.set_option and graphscope.get_option API

## Related issue number
Fixes #7 
Fixes #27

### CODE REVIEW

File: coordinator/gscoordinator/utils.py [hunk 1]
```diff
@@ -293,28 +294,29 @@ def compile_graph_frame(
             f.write(content)
 
     # compile
+    logger.info("Building graph ...")
```
Comment (sighingnow) [inline coordinator/gscoordinator/utils.py:297]: “Building graph ...” -> "Building graph binding..." or "Building graph library...".

File: python/graphscope/client/session.py [hunk 2]
```diff
@@ -211,21 +206,25 @@ def __init__(
...
                 Defaults to 600.
 
-
             k8s_waiting_for_delete (bool, optional): Waiting for service delete or not. Defaults to False.
 
             **kw (dict, optional): Other optional parameters will be put to :code:`**kw`.
-                If your kubernetes cluster deployed on inner virtual machine
+                - If your kubernetes cluster deployed on inner virtual machine
                 (such as minikube with param --vm-driver is not None), you can specify
                 :code:`k8s_minikube_vm_driver` is :code:`True`.
```
Comment (sighingnow) [inline python/graphscope/client/session.py:214]: Format:

```python
- k8s_minikube_vm_driver (bool, optional): If your .....
```

File: python/graphscope/client/session.py [hunk 3]
```diff
@@ -211,21 +206,25 @@ def __init__(
...
                     eg: {"config_file": "~/.kube/config", "context": None, "persist_config": True}
                     config_file: Name of the kube-config file.
                     context: set the active context. If is set to None, current_context from config file will be used.
                     persist_config: If True, config file will be updated when changed(e.g GCP token refresh).
 
+                - log_level: Deprecated,
+                move this param as a global configuration. Set via `graphscope.set_option(log_level='DEBUG')`
+
+                - show_log: Deprecated,
+                mmove this param as a global configuration.Set via `graphscope.set_option(show_log=True)`
```
Comment (sighingnow) [inline python/graphscope/client/session.py:227]: typo: `mmove` -> `move`.

File: python/graphscope/client/session.py [hunk 4]
```diff
@@ -883,7 +865,11 @@ def learning(self, graph, nodes=None, edges=None, gen_labels=None):
         return learning_graph
 
 
-session = Session
+def session_impl(*args, **kwargs):
+    return Session(*args, **kwargs)
+
+
+session = functools.partial(session_impl)
```
Comment (sighingnow) [inline python/graphscope/client/session.py:872]: That is bad, now `help(graphscope.session)` shows nothing.

File: python/graphscope/client/utils.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            formatter = ConditionalFormatter('%(asctime)s %(levelname)s - %(message)s')
+            handler.setFormatter(formatter)
+            logger = logging.getLogger("graphscope")
+            logger.setLevel("INFO")
+            logger.addHandler(handler)
+            logger.info("with formatting")
+            # 2020-12-21 13:44:52,537 INFO - with formatting
+            logger.info("without formatting", extra={'simple': True})
+            # without formatting
+    """
```
Comment (sighingnow) [inline python/graphscope/client/utils.py:44]: Please double check the sphinx format in generated docs.

**Review by sighingnow — DISMISSED** [review]

File: python/graphscope/client/utils.py (hunk 5 above)
Comment (sighingnow) [reply python/graphscope/client/utils.py:44]: You could have such code block like:


```python
.. code:: python

    >>> import logging
    >>> import sys
    >>> handler = logging.StreamHandler(sys.stdout)
    >>> formatter = ConditionalFormatter('%(asctime)s %(levelname)s - %(message)s')
    >>> handler.setFormatter(formatter)
    >>> logger = logging.getLogger("graphscope")
    >>> logger.setLevel("INFO")
    >>> logger.addHandler(handler)
    >>> logger.info("with formatting")
    2020-12-21 13:44:52,537 INFO - with formatting
    >>> logger.info("without formatting", extra={'simple': True})
    without formatting
```

**Comment by lidongze0629** [comment]
Thanks, @sighingnow, I have improved the coding following your advice.

**Review by yecol — APPROVED** [review]
LGTM.

**Comment by sighingnow** [comment]
There are still many `show_logs=True/False` in test_xxxxx.py.

**Pull request merged**